packages/core/bootstrap/src/lib/ws/index.ts

```diff
@@ -38,24 +38,34 @@ export const withWebSockets =
       return await awaitResult(context, input, deadline)
     }
 
-    await separateBatches(input, async (singleInput: AdapterRequest) => {
-      const subscriptionMsg = wsHandler.subscribe(singleInput)
-      if (!subscriptionMsg) return
-      const subscriptionPayload: WSSubscriptionPayload = {
-        connectionInfo: {
-          key: wsConfig.connectionInfo.key,
-          url: wsHandler.connection.url,
-        },
-        subscriptionMsg,
-        input: singleInput,
-        context,
-      }
+    if (isConnected(store, wsConfig.connectionInfo.key)) {
+      await separateBatches(input, async (singleInput: AdapterRequest) => {
```

Comment:
Since this is the `subscribeReadyEpic` repeated could we pull this out of here and only have it in epics?

Reply:
We'll address it in this PR https://app.clubhouse.io/chainlinklabs/story/16732/refactor-ws-middleware-flow